src/librustc_mir/util/liveness.rs

```diff
@@ -40,25 +40,48 @@ use rustc_data_structures::indexed_set::IdxSetBuf;
 use rustc_data_structures::work_queue::WorkQueue;
 use util::pretty::{dump_enabled, write_basic_block, write_mir_intro};
 use rustc::ty::item_path;
+use rustc::mir::Local;
 use rustc::mir::visit::MirVisitable;
 use std::path::{Path, PathBuf};
 use std::fs;
 use rustc::ty::TyCtxt;
 use std::io::{self, Write};
 use transform::MirSource;
 
-pub type LocalSet = IdxSetBuf<Local>;
+pub type LocalSet<V: LiveVariableMap> = IdxSetBuf<V::LiveVar>;
```

Comment:
is this giving a warning? If so, change `V::LiveVar` to `<V as LiveVariableMap>::LiveVar` and remove the bound (for better or worse, this compiles)